### PR TITLE
Added short-benchmarks for cumulus

### PR DESCRIPTION
**Pull request by bkontur**
This PR adds new jobs for `short-benchmarks` for system parachain runtimes in cumulus subdirectory which was added to the Cumulus repo [here](https://github.com/paritytech/cumulus/pull/2846).

### CODE REVIEW

File: .gitlab/pipeline/short-benchmarks.yml [hunk 1]
```diff
@@ -24,3 +27,58 @@ short-benchmark-westend:
   <<: *short-bench
   variables:
     RUNTIME: westend
+
+# run short-benchmarks for system parachain runtimes from cumulus
+
+.short-benchmark-cumulus: &short-bench-cumulus
```
Comment (bkontur) [inline .gitlab/pipeline/short-benchmarks.yml:33]: `.short-benchmark-cumulus` and `.short-benchmark-polkadot` could be possibly merged to `.short-benchmark-template` with some variables parametrization, do we want to do it?

File: .gitlab/pipeline/build.yml [hunk 2]
```diff
@@ -139,7 +139,7 @@ build-short-benchmark:
     - .run-immediately
     - .collect-artifacts
   script:
-    - cargo build --profile release --locked --features=runtime-benchmarks
+    - cargo build --profile release --locked --features=runtime-benchmarks --bin polkadot
     - mkdir -p artifacts
     - target/release/polkadot --version
     - cp ./target/release/polkadot ./artifacts/
```
Comment (bkontur) [inline .gitlab/pipeline/build.yml:145]: There is also `mv ./target/testnet/polkadot ./artifacts/.` in the step `build-linux-stable`, so this step overrides that, right?
Possibly we could do something like: `cp ./target/release/polkadot ./artifacts/short-benchmarks/`.

Also the same for `polkadot-parachain` binary [build-linux-stable-cumulus](https://github.com/paritytech/polkadot-sdk/blob/master/.gitlab/pipeline/build.yml#L165) does `mv ./target/release/polkadot-parachain ./artifacts/.` but also `build-short-benchmark-cumulus` does `cp ./target/release/polkadot-parachain ./artifacts/.

I am not missing anything, there is a possible consequence that later jobs could use binary with `--features runtime-benchmarks`.

**Review by bkchr — APPROVED** [review]

**Review by alvicsam — APPROVED** [review]

**Comment by gilescope** [comment]
Given that it's short, could we do it all in one job rather than giving the CI another 10+ jobs to handle? In fact could we do the relay chain ones in the same job (also rococo on the relay jobs is currently missed out fyi)?

**Comment by bkontur** [comment]
> Given that it's short, could we do it all in one job rather than giving the CI another 10+ jobs to handle? In fact could we do the relay chain ones in the same job (also rococo on the relay jobs is currently missed out fyi)?

I think separate jobs per runtime is much more clear or transparent.
They all reuse the same prebuild binaries, so the execution itself is "fast".
They can run in "parallel" (depends on free vm runners I guess).
You can see exactly which runtime has problem, so it is easier for identification.
You can even see, which runtimes are configured and which not.

To Roocco, I dont know, it was missing also in old polkadot repo.

File: .gitlab/pipeline/short-benchmarks.yml [hunk 3]
```diff
@@ -24,3 +27,58 @@ short-benchmark-westend:
...
+  stage: short-benchmarks
+  extends:
+    - .common-refs
+    - .docker-env
+  needs:
+    - job: build-short-benchmark-cumulus
+      artifacts: true
+  variables:
+    RUNTIME_CHAIN: benchmarked-runtime-chain
```
Comment (alvicsam) [inline .gitlab/pipeline/short-benchmarks.yml:42]:
```suggestion
    RUNTIME_CHAIN: benchmarked-runtime-chain
  tags:
    - benchmark
```

**Review by alvicsam — CHANGES_REQUESTED** [review]
@gilescope is correct. Lets use another runners for these benchmarks

File: .gitlab/pipeline/short-benchmarks.yml (hunk 3 above)
Comment (bkontur) [reply .gitlab/pipeline/short-benchmarks.yml:42]: @alvicsam ok, but I dont know how, can you help with that?
or that suggested change is the fix?

Comment (alvicsam) [reply .gitlab/pipeline/short-benchmarks.yml:42]: yep, the suggested fix will do that

Comment (bkontur) [reply .gitlab/pipeline/short-benchmarks.yml:42]: @alvicsam do we need to add that tag also to [short-benchmark-polkadot](https://github.com/paritytech/polkadot-sdk/pull/1183/files#diff-37ac91a4b108440e9511bd13341ff237deebb3cf99815466cafe8b4468939e7bR8-R17)?

Comment (bkontur) [reply .gitlab/pipeline/short-benchmarks.yml:44]: ok, fixed

Comment (alvicsam) [reply .gitlab/pipeline/short-benchmarks.yml:44]: thank you!

**Review by alvicsam — APPROVED** [review]

**Pull request merged**